content/4_Gold/PURS.mdx

```diff
@@ -739,28 +802,40 @@ Covered in [Platinum](/plat/seg-ext).
 
 <CPPSection>
 
-Using indexed set, we can solve this in just a few lines.
+Using an indexed set, we can solve this in just a few lines.
 
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
 #include <ext/pb_ds/assoc_container.hpp>
 using namespace __gnu_pbds;
-template <class T> using Tree = tree<T, null_type, less<T>,
-	rb_tree_tag, tree_order_statistics_node_update>;
+template <class T> using Tree = tree<
+	T, null_type, less<T>,
+	rb_tree_tag, tree_order_statistics_node_update
+>;
 
 int main() {
-	int T; cin >> T;
-	for (int i = 0; i < T; ++i) {
-		int n; cin >> n;
-		Tree<int> TS; long long numInv = 0;
-		for (int j = 0; j < n; ++j) {
-			int x; cin >> x;
-			numInv += j-TS.order_of_key(x); // gives # elements before it > x
-			TS.insert(x);
+	int test_num;
+	cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		Tree<int> tree;
+		
+		int arr_len;
+		cin >> arr_len;
+		long long inv_num = 0;
+		for (int i = 0; i < arr_len; i++) {
+			int x;
+			cin >> x;
+			/*
+			 * .order_of_key(x) gives the # of elements
+			 * in the tree that are greater than x.
```

Comment:
strictly less than right?

Reply:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/55369003/185267417-a6116e41-2356-415a-8735-83fa2267ccfc.png">
this is the OG comment

Reply:
<img width="720" alt="image" src="https://user-images.githubusercontent.com/81596991/185267494-7f1ecc9c-6897-421a-8d0a-177544bb3a02.png">

Reply:
wait lemme reread this

Reply:
oh yeah you're right

Reply:
actually no i think you were, the og comment was just misleading as frick

Reply:
the problem is asking for something different though?

Reply:
we want the # of inversions
so for each element, we want the # of elements preceding it that are greater than said element

Reply:
thought maybe they passed in some cryptic stuff in the type definition to make it go the other way

Reply:
OOPS LMAO I THOUGHT IT WAS JUST order_of_key and not the i - stuff